src/families/celo/hw-app-celo.ts

```diff
@@ -0,0 +1,80 @@
+import Eth from "@ledgerhq/hw-app-eth";
+import { splitPath, foreach } from "@ledgerhq/hw-app-eth/lib/utils";
+import { decode, encode } from "rlp";
+
+/**
+ * Copied from celo-web-wallet, which in turn reused hw-app-eth/src/Eth.js
+ * https://github.com/celo-tools/celo-web-wallet/blob/master/src/features/ledger/CeloLedgerApp.ts
+ */
+export class CeloApp extends Eth {
```

Comment:
I'm not sure this is really a good idea to extend `@ledgerhq/hw-app-eth`

any idea what is the reason to extend it in this context? also why ethereum (it's a fork?) and any reason in that case why the "hw-app-eth" itself wouldn't fit?   (thinking out loud here)

Reply:
I agree, there's a small risk things might stop working when hw-app-eth changes.

The reason for extending it is that yes, Celo is an Ethereum fork but it does have a different form of transactions. I went with your line of thought at first and did try to use `hw-app-eth`, but the differences in transactions mean that most of the code is same/similar, but we have to encode a couple more fields:

https://docs.celo.org/developer-guide/celo-for-eth-devs#key-differences-between-celo-and-ethereum
`Celo transaction objects are slightly different from transaction objects on Ethereum. Ethereum transaction objects include fields to, value, gas, gasPrice, data, nonce, signature (v,r,s). Celo transaction objects include the same fields as Ethereum transaction objects, plus feeCurrency, gatewayFee and gatewayFeeRecipient`

If we don't want to extend `@ledgerhq/hw-app-eth`, I can copy paste it and overwrite relevant pieces or stick to a certain version of hw-app-eth. Any other ideas?

Reply:
after double thinking on this topic. I would prefer we don't use inheritence, especially that we're planning some evolution on the ETH signTransaction signature. You can just take the part you need and fork the code.

Reply:
Sure, will do 👍 
I'll add a spec as well

Reply:
ok, that's done @gre